SubscribeToMailchimp.module

```diff
@@ -137,16 +137,91 @@ class SubscribeToMailchimp extends WireData implements Module, ConfigurableModul
                 $f->attr('value', $data[$name]);
             $form->add($f);
         }
-        foreach($checkboxfields as $name => $label) {       
+        foreach($checkboxfields as $name => $label) {
             $f = wire('modules')->get("InputfieldCheckbox");
             $f->name = $name;
             $f->label = $label;
             $f->attr('checked', empty($data[$name]) ? '' : 'checked');
             $f->columnWidth = 100;
             $form->add($f);
         }
-        $wrap->add($form);   
+
+        // TEST SETTINGS ||>>>
+        $f = wire('modules')->get('InputfieldCheckbox');
+        $f->attr('name', '_mailchimp_test_settings');
+        $f->label = __('Test your current settings!');
+        $f->description = __('Check this box and press save, to test the API Key and default list settings!');
+        $f->attr('value', 1);
+        $f->attr('checked', '');
+        $f->icon = 'heartbeat';
+        $f->columnWidth = wire('session')->mailchimp_test_settings ? 40 : 100;
+        $form->add($f);
+
+        if(wire('session')->mailchimp_test_settings) {
+            wire('session')->remove('mailchimp_test_settings');
+            $response = self::testSettings();
+            $f = wire('modules')->get('InputfieldMarkup');
+            $f->attr('name', '_test_response');
+            $f->label = __('Test Response: ');
+            $f->columnWidth = 60;
+            $f->attr('value', $response);
+            $form->add($f);
+        } else if(wire('input')->post->_mailchimp_test_settings) {
+            wire('session')->set('mailchimp_test_settings', 1);
+        }
+        // <<<|| TEST SETTINGS
+
+        $wrap->add($form);
         return $wrap;
     }
+    // static, because mainly it gets called from static function getModuleConfigInputfields!
+    // But also can get called from within your template files: $response = SubscribeToMailchimp::testSettings($listID);
+    public static function testSettings($list = null) {
+        $mc = wire('modules')->get('SubscribeToMailchimp');
+        $list = is_null($list) ? $mc->default_list : $list;
+        $dc = explode('-',$mc->api_key)[1]; // get the datacenter code from the api key, to use it in the api url
+        $request = "https://{$dc}.api.mailchimp.com/3.0/lists/{$list}/";
```

Comment:
Yep, at first I used it, but it returns to much. I only need the base, but your method also returns ```/{$list}/members/```. Maybe you can modify the getApiBase() method with an optional param, so that it returns a base URL, with optionally added string(s) to it:
``` 
private function getApiBase($list = null, $param = null) {
    $dc = explode('-',$this->api_key)[1];
    $return = "https://".$dc.".api.mailchimp.com/3.0/";
    if(!is_null()) $return .= "lists/".$list."/";
    if(!is_null($param)) $return .= $param; // should we automatically add a trailing slash here ?
    return $return;
}
 ```